### ISSUE TITLE
Fatai IO Error 104

### ISSUE THREAD

**Issue by MostlyKIGuess** [RTimothyEdwards/magic#354]
![image](https://github.com/user-attachments/assets/b2df7f01-f222-4fa9-800f-8974cf7e43bd)

- I tried opening with -noconsole but no luck sadly, and it just crashed while I was extending metal2, it happened out of the blue.
- Not sure how to reconstruct.
### OS Information:
- Arch Linux
- Magic Version: Latest

**Comment by MostlyKIGuess** [comment]
I was running on Wayland and had TCL , TK lib installed.

**Comment by RTimothyEdwards** [comment]
The issue here, `couldn't connect to display ":1"`, generally means that there isn't an X11 server running.  It seems very weird to me that it would happen in the middle of running the program.  It's like the X server crashed or something.  You might want to try running magic with `-d XR` (requires installing cairo graphics library) which would avoid making low-level X11 calls.

Sorry about the delay in responding.  I had to get through a tape-out and into the holiday before I had enough time to respond to emails and issues from the last couple of months.

**Comment by RTimothyEdwards** [comment]
My best guess is that this is probably a bug in the X11 implementation in Wayland, although I cannot discount the possibility of a bug in magic's X11 calls, especially around the interactive wiring and copying pixmaps rapidly between the application and backing store.  Hopefully using cairo or OpenGL graphics avoids the problem altogether.

**Comment by MostlyKIGuess** [comment]
> My best guess is that this is probably a bug in the X11 implementation in Wayland, although I cannot discount the possibility of a bug in magic's X11 calls, especially around the interactive wiring and copying pixmaps rapidly between the application and backing store. Hopefully using cairo or OpenGL graphics avoids the problem altogether.

Yes it somehow is connected to handling of the tech files btw, for some reason using the same tech file updating via GUI would not consider it and not update the corresponding metal similar to terminal as well, I would run the command twice or thrice and it would work giving proper DRC checks and correct metals for the files. Would you happen to know about this issue? Perhaps if there's any lead I would love to work on it
